testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6218,10 +6226,43 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
         isWithin(9.0, 0.000001));
     f.checkScalarApprox("log(cast(10e-3 as real), 10)", "DOUBLE NOT NULL",
         isWithin(-2.0, 0.000001));
+    f.checkScalarApprox("log(2.0/3, 100)", "DOUBLE NOT NULL",
+        isWithin(-0.08804562952784059, 0.000001));
+    f.checkScalarApprox("log(0.5, 2)", "DOUBLE NOT NULL",
+        isWithin(-1.0, 0.000001));
```

Comment:
no tests for negative/zero?

Reply:
Regarding log10 and log functions, I will not fix them in this jira, because the jira is only for log2 functions